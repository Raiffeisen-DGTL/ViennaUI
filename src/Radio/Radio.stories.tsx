import React from 'react';
import { Story, Meta } from 'storybook';
import { Radio, RadioProps } from './Radio';
import { Groups } from '../Groups';

export default {
    title: 'Development/Radio',
    component: Radio,
} as Meta;

export const Overview: Story<RadioProps> = (args) => {
    const [state, setState] = React.useState('');

    return (
        <Groups>
            {Array.from(Array(4)).map((_, i) => {
                const id = (i + 1).toString();

                return (
                    <Radio
                        key={id}
                        checked={state === id}
                        name='group1'
                        value={id}
                        onChange={({ value }) => setState(value)}>
                        {'Radio #' + id}
                    </Radio>
                );
            })}
        </Groups>
    );
};

export const WithAdaptive: Story<RadioProps> = (args) => {
    return (
        <Radio disabled size={{ m: 'm', s: 'l' }} {...args}>
            Check
        </Radio>
    );
};

WithAdaptive.storyName = 'Адаптив';

export const ViewOnly: Story<RadioProps> = (args) => {
    return (
        <>
            <Groups>
                {Array.from(Array(4)).map((_, i) => {
                    const id = (i + 1).toString();

                    return (
                        <Radio viewOnly key={id} checked={'2' === id} name='group1' value={id}>
                            {'Radio #' + id}
                        </Radio>
                    );
                })}
            </Groups>
            <Groups>
                {Array.from(Array(4)).map((_, i) => {
                    const id = (i + 1).toString();
                    const checked = '2' === id;
                    if (checked) {
                        return (
                            <Radio viewOnly key={id} checked={checked} name='group1' value={id}>
                                {'Radio #' + id}
                            </Radio>
                        );
                    }
                    return '';
                })}
            </Groups>
            <Groups>
                {Array.from(Array(4)).map((_, i) => {
                    const id = (i + 1).toString();
                    const checked = '2' === id;
                    if (checked) {
                        return (
                            <Radio viewOnly viewOnlyDisableIcon key={id} checked={checked} name='group1' value={id}>
                                {'Radio #' + id}
                            </Radio>
                        );
                    }
                    return '';
                })}
            </Groups>
        </>
    );
};

export const RadioInvalid: Story<RadioProps> = (args) => {
    const [state, setState] = React.useState('');

    return (
        <Groups>
            {Array.from(Array(4)).map((_, i) => {
                const id = (i + 1).toString();

                return (
                    <Radio
                        key={id}
                        checked={state === id}
                        invalid
                        name='group1'
                        value={id}
                        onChange={({ value }) => setState(value)}>
                        {'Radio #' + id}
                    </Radio>
                );
            })}
        </Groups>
    );
};
