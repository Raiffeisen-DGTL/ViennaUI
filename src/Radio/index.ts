export { Radio, type RadioEvent, type RadioProps, type RadioSize } from './Radio';
