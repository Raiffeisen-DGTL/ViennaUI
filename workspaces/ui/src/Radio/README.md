# Radio

Компонент выбора одного из множества значений

## Импорт

```
import { Hint } from 'vienna-ui';
```

## Свойства / Props

| Prop | Type | Default | Description |
| --- | --- | --- | --- |
| children | ReactNode | false | Отображаемый лейбл |
| ref | ((instance: HTMLInputElement \| null) => void) \| RefObject \| null \| undefined | false | Сcылка на нативный элемент input, доступна после отрисовки |
| size | "s" \| "m" \| "l" \| undefined | false | Размеры |
| value\* | string | false | Значение компонента |
| invalid | boolean \| undefined | false | Компонент отображается как ошибочный если true |
| onBlur | RadioEvent<FocusEvent<HTMLInputElement>> \| undefined | false | Обработчик события при потере фокуса компонентом |
| onFocus | RadioEvent<FocusEvent<HTMLInputElement>> \| undefined | false | Обработчик события при получении фокуса компонентом |
| onChange | RadioEvent<FocusEvent<HTMLInputElement>> \| undefined | false | Обработчик события при клике на компонент |
| checked | boolean \| undefined | false | Статус выбранного значения |
| name | string \| undefined | false |
| disabled | boolean \| undefined | false |

## Использование

> Компонент является контролируемым, то есть чтобы отобразить выбранные элементы, необходимо получить их значение через обработчик `onChange` и прокинуть проверку в `checked`.

```{() => {
    const [state, setState] = React.useState({ value: '2' });
    return (
        <Groups>
            <Radio
                checked={state.value === '1'}
                name='group1'
                value='1'
                onChange={(e, { value }) => setState({ value })}>
                Radio #1
            </Radio>
            <Radio
                checked={state.value === '2'}
                name='group1'
                value='2'
                onChange={(e, { value }) => setState({ value })}>
                Radio #2
            </Radio>
            <Radio
                checked={state.value === '3'}
                name='group1'
                value='3'
                onChange={(e, { value }) => setState({ value })}>
                Radio #3
            </Radio>
            <Radio
                checked={state.value === '4'}
                name='group1'
                value='4'
                onChange={(e, { value }) => setState({ value })}>
                Radio #4
            </Radio>
        </Groups>
    );
}}
```

## Выбранный элемент

##### Свойство `checked`

```
<Radio checked>
    Content
</Radio>
```

## Размеры

##### Свойство `size`

Компонент имеет стандартные размеры `s`, `m` (по умолчанию) и `l`.

## Состояния

##### Свойство `disabled`, `invalid`

Компонент имеет состояние `disabled` и `invalid`

```
<Radio
    disabled
    checked
>
    Content
</Radio>
<Radio
    invalid
    checked
>
    Content
</Radio>
```

## Значение

##### Свойство `value`

Значение, которое передается в компонент и потом возвращается `onChange`.

```
<Radio
    value='value-1'
    checked={state === 'value-1'}
>
    Content
</Radio>
```
