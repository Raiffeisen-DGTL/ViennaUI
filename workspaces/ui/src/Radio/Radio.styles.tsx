import styled, { css } from 'styled-components';
import { getPresets } from 'vienna.ui-primitives';

const radio = getPresets('radio', {
    base: null,
    size: 'size',
    hover: null,
    disabled: null,
    invalid: null,
    checked: null,
});

const label = getPresets('radio.label', {
    base: null,
    size: 'size',
    disabled: null,
});

const checkmark = getPresets('radio.checkmark', {
    base: null,
    disabled: null,
});

export const Icon = styled.span<{ size: any; disabled: any; invalid?: boolean }>`
    display: block;
    position: relative;
    flex: 0 0 auto;
    box-sizing: border-box;
    transition: all .2s;
    align-self: flex-start;

    ${radio.base}
    ${radio.size}

    ${({ disabled }): any =>
        disabled &&
        css`
            ${radio.disabled}
        `}

    ${({ invalid }): any =>
        invalid &&
        css`
            ${radio.invalid}
        `}
`;

export const Content = styled.span<{ size: any }>`
    ${label.base}
    ${label.size}
`;

export const Box = styled.label<{ size: any; disabled: any }>`
    padding: 2px 0;
    display: inline-flex;
    align-items: center;
    transition: all 0.2s;

    &:hover ${Icon} {
        ${radio.hover}
        ${({ disabled }): any =>
            disabled &&
            css`
                ${radio.disabled}
            `}
    }

    ${Content} {
        ${({ disabled }): any =>
            disabled &&
            css`
                ${label.disabled}
            `}
    }
`;

export const Input = styled.input<{ invalid?: boolean }>`
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    margin: 0;
    opacity: 0;
    z-index: -1;

    &:checked ~ ${Icon} {
        ${radio.checked}

        ${({ disabled }): any =>
            disabled &&
            css`
                ${radio.disabled}
            `}

        ${({ invalid }): any =>
            invalid &&
            css`
                ${radio.invalid}
            `}

        &::after {
            content: '';
            display: none;
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
            width: 50%;
            height: 50%;
            border-radius: 50%;

            display: block;
            ${checkmark.base}

            ${({ disabled }): any =>
                disabled &&
                css`
                    ${checkmark.disabled}
                `};
        }
    }

    &:focus ~ ${Icon} {
        ${radio.hover}
    }
`;
