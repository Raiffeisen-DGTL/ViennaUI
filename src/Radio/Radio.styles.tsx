import styled, { css } from 'styled-components';
import { getPresets } from '../Utils/styling';
import { Breakpoints, responsivePreset } from '../Utils/responsiveness';
import { RadioSize } from './Radio';

const radio = getPresets('radio', {
    base: null,
    size: responsivePreset('$size', 'm'),
    hover: null,
    disabled: null,
    invalid: null,
    checked: null,
});

const label = getPresets('radio.label', {
    base: null,
    size: responsivePreset('$size', 'm'),
    disabled: null,
});

const checkmark = getPresets('radio.checkmark', {
    base: null,
    disabled: null,
});

interface PropsIcon<B = Breakpoints> {
    $size?: RadioSize<B>;
    $disabled?: boolean;
    $invalid?: boolean;
}
export const Icon = styled.span<PropsIcon>`
    display: block;
    position: relative;
    flex: 0 0 auto;
    box-sizing: border-box;
    transition: all 0.2s;
    align-self: flex-start;

    ${radio.base}
    ${radio.size}

    ${({ $disabled }) =>
        $disabled &&
        css`
            ${radio.disabled}
        `}

    ${({ $invalid }) =>
        $invalid &&
        css`
            ${radio.invalid}
        `}
`;

interface PropsContent<B = Breakpoints> {
    $size?: RadioSize<B>;
}
export const Content = styled.span<PropsContent>`
    ${label.base}
    ${label.size}
`;

export interface PropsBox<B = Breakpoints> {
    $size?: RadioSize<B>;
    $disabled?: boolean;
}
export const Box = styled.label<PropsBox>`
    position: relative;
    padding: 2px 0;
    display: inline-flex;
    align-items: center;
    transition: all 0.2s;

    &:hover ${Icon} {
        ${radio.hover}
        ${({ $disabled }) =>
            $disabled &&
            css`
                ${radio.disabled}
            `}
    }

    ${Content} {
        ${({ $disabled }) =>
            $disabled &&
            css`
                ${label.disabled}
            `}
    }
`;

interface PropsInput<B = Breakpoints> {
    $size?: RadioSize<B>;
    $disabled?: boolean;
    $invalid?: boolean;
}
export const Input = styled.input<PropsInput>`
    position: absolute;
    top: 0;
    left: 0;
    height: 100%;
    width: 100%;
    margin: 0;
    opacity: 0;
    z-index: -1;

    &:checked ~ ${Icon} {
        ${radio.checked}

        ${({ $disabled }) =>
            $disabled &&
            css`
                ${radio.disabled}
            `}

        ${({ $invalid }) =>
            $invalid &&
            css`
                ${radio.invalid}
            `}

        &::after {
            content: '';
            position: absolute;
            top: 50%;
            left: 50%;
            transform: translate(-50%, -50%);
            width: 50%;
            height: 50%;
            border-radius: 50%;

            display: block;
            ${checkmark.base}

            ${({ $disabled }) =>
                $disabled &&
                css`
                    ${checkmark.disabled}
                `};
        }
    }

    &:focus ~ ${Icon} {
        ${radio.hover}
    }
`;
