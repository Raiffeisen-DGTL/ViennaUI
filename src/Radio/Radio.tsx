import React, { useCallback, ForwardedRef, FC } from 'react';
import { Box, PropsBox, Input, Icon, Content } from './Radio.styles';
import { ResponsiveProp, Breakpoints } from '../Utils/responsiveness';
import { BoxStyled } from '../Utils/styled';

export type RadioSize<B = Breakpoints> = ResponsiveProp<'s' | 'm' | 'l', B>;

export type RadioEvent<T> = (event: T, data?: { name?: string; value: string }) => void;

export interface RadioProps<B = Breakpoints>
    extends Omit<BoxStyled<typeof Box, PropsBox>, 'onFocus' | 'onBlur' | 'onChange'> {
    /** Размеры */
    size?: ResponsiveProp<'s' | 'm' | 'l', B>;
    /** Имя компонента (важно для группировки) */
    name?: string;
    /** Значение компонента */
    value: string;
    /** Состояние компонента */
    checked?: boolean;
    /** Компонент неактивен если true */
    disabled?: boolean;
    /** Компонент отображается как ошибочный если true */
    invalid?: boolean;
    /** Обработчик события при потере фокуса компонентом  */
    onBlur?: RadioEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при получении фокуса компонентом  */
    onFocus?: RadioEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при клике на компонент  */
    onChange?: RadioEvent<React.ChangeEvent<HTMLInputElement>>;
}

function RadioInternal<B = void>(
    {
        children,
        size = 'm',
        checked = false,
        disabled = false,
        invalid,
        name,
        value,
        tabIndex,
        onChange,
        onBlur,
        onFocus,
        ...attrs
    }: RadioProps<B>,
    ref: React.Ref<HTMLInputElement>
) {
    const handleEvent = useCallback(
        (callback: RadioEvent<any> | undefined) => {
            return (event: React.FormEvent<HTMLInputElement>) => {
                if (!disabled && typeof callback === 'function') {
                    callback(event, { name, value });
                }
            };
        },
        [disabled, name, value]
    );

    return (
        <Box {...(attrs as {})} $size={size} $disabled={disabled}>
            <Input
                ref={ref}
                type='radio'
                name={name}
                value={value}
                tabIndex={tabIndex}
                $size={size}
                disabled={disabled}
                $disabled={disabled}
                $invalid={invalid}
                aria-invalid={!!invalid}
                checked={checked}
                onChange={handleEvent(onChange)}
                onBlur={handleEvent(onBlur)}
                onFocus={handleEvent(onFocus)}
            />
            <Icon $size={size} $disabled={disabled} $invalid={invalid} />
            {children ? <Content $size={size}>{children}</Content> : null}
        </Box>
    );
}

export const Radio = React.forwardRef(RadioInternal) as <B = Breakpoints>(
    props: RadioProps<B> & { ref?: ForwardedRef<HTMLInputElement> }
) => ReturnType<typeof RadioInternal>;

(Radio as FC).displayName = 'Radio';
