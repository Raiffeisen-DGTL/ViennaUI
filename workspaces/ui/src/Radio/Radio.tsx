import React, { useCallback, HTMLAttributes } from 'react';
import { Box, Input, Icon, Content } from './Radio.styles';

export type RadioEvent<T> = (event: T, data?: { name?: string; value: string }) => void;
export interface RadioProps extends Omit<HTMLAttributes<HTMLLabelElement>, 'size' | 'onChange' | 'onBlur' | 'onFocus'> {
    /** Отображаемый лейбл */
    children?: React.ReactNode;
    /** Сcылка на нативный элемент input, доступна после отрисовки */
    ref?: React.Ref<HTMLInputElement>;
    /** Размеры */
    size?: 's' | 'm' | 'l';
    /** Имя компонента (важно для группировки) */
    name?: string;
    /** Значение компонента */
    value: string;
    /** Состояние компонента */
    checked?: boolean;
    /** Компонент неактивен если true */
    disabled?: boolean;
    /** Компонент отображается как ошибочный если true */
    invalid?: boolean;
    /** Обработчик события при потере фокуса компонентом  */
    onBlur?: RadioEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при получении фокуса компонентом  */
    onFocus?: RadioEvent<React.FocusEvent<HTMLInputElement>>;
    /** Обработчик события при клике на компонент  */
    onChange?: RadioEvent<React.ChangeEvent<HTMLInputElement>>;
}

export const Radio: React.FC<RadioProps> = React.forwardRef(
    (props: React.PropsWithChildren<Omit<RadioProps, 'ref'>>, ref: React.Ref<HTMLInputElement>): JSX.Element => {
        const {
            children,
            size = 'm',
            checked = false,
            disabled = false,
            invalid,
            name,
            value,
            tabIndex,
            onChange,
            onBlur,
            onFocus,
            ...attrs
        } = props;

        const handleEvent = useCallback(
            (callback: RadioEvent<any> | undefined) => {
                return (event: React.FormEvent<HTMLInputElement>) => {
                    if (!disabled && typeof callback === 'function') {
                        callback(event, { name, value });
                    }
                };
            },
            [disabled, name, value]
        );

        return (
            <Box size={size} disabled={disabled} {...attrs}>
                <Input
                    ref={ref}
                    type='radio'
                    name={name}
                    value={value}
                    tabIndex={tabIndex}
                    disabled={disabled}
                    invalid={invalid}
                    aria-invalid={!!invalid}
                    checked={checked}
                    onChange={handleEvent(onChange)}
                    onBlur={handleEvent(onBlur)}
                    onFocus={handleEvent(onFocus)}
                />
                <Icon size={size} disabled={disabled} invalid={invalid} />
                {children ? <Content size={size}>{children}</Content> : null}
            </Box>
        );
    }
);

Radio.defaultProps = {
    size: 'm',
    checked: false,
};

Radio.displayName = 'Radio';
export default Radio;
