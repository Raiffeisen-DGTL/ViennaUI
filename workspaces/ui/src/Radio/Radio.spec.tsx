import React from 'react';
import Radio, { RadioProps } from './Radio';

test('Radio', () => {
    const snap = snapshot.render(<Radio value='x' />);
    expect(snap).toMatchSnapshot();
});

test('Radio w/ label', () => {
    const snap = snapshot.render(<Radio value='x'>Label</Radio>);
    expect(snap).toMatchSnapshot();
});

test('Radio w/ name', () => {
    const snap = snapshot.render([
        <Radio key='1' name='Radio1' value='x'>
            Label
        </Radio>,
        <Radio key='2' name='Radio2' value='y'>
            Label
        </Radio>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Radio w/ checked', () => {
    const snap = snapshot.render([
        <Radio key='1' name='Radio1' value='x' checked>
            Label
        </Radio>,
        <Radio key='2' name='Radio2' value='y'>
            Label
        </Radio>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Radio w/ disabled', () => {
    const snap = snapshot.render([
        <Radio key='1' name='Radio1' value='x' disabled>
            Label
        </Radio>,
        <Radio key='2' name='Radio2' value='y'>
            Label
        </Radio>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Radio w/ invalid', () => {
    const snap = snapshot.render(
        <Radio key='1' name='Radio1' value='x' invalid>
            Label
        </Radio>
    );

    expect(snap).toMatchSnapshot();
});

test('Radio w/ checked and disabled', () => {
    const snap = snapshot.render([
        <Radio key='1' name='Radio1' value='x' checked disabled>
            Label
        </Radio>,
        <Radio key='2' name='Radio2' value='y'>
            Label
        </Radio>,
    ]);

    expect(snap).toMatchSnapshot();
});

test('Radio w/ size', () => {
    const sizes: RadioProps['size'][] = ['s', 'm', 'l'];

    sizes.forEach((size) => {
        const snap = snapshot.render(
            <Radio value='x' size={size}>
                Label
            </Radio>
        );

        expect(snap).toMatchSnapshot(size);
    });
});

test('Radio w/ any props', () => {
    const props = {
        className: 'className',
        id: 'id',
        title: 'title',
        onClick: () => 'Click',
    };

    const component = (
        <Radio {...props} name='name' value='x'>
            Label
        </Radio>
    );

    const snap = snapshot.render(component);
    expect(snap).toMatchSnapshot();

    const wrapper = snapshot.shallow(component);
    for (const [key, value] of Object.entries(props)) {
        expect(wrapper.props()[key]).toEqual(value);
    }
});

test('Radio w/ onBlur()', () => {
    const fn = jest.fn();
    const event = {};

    const snap = snapshot.shallow(
        <Radio name='radio' value='x' onBlur={fn}>
            Label
        </Radio>
    );

    snap.childAt(0).simulate('blur', event);
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'radio', value: 'x' });
});

test('Radio w/ onFocus()', () => {
    const fn = jest.fn();
    const event = {};

    const snap = snapshot.shallow(
        <Radio name='radio' value='x' onFocus={fn}>
            Label
        </Radio>
    );

    snap.childAt(0).simulate('focus', event);
    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'radio', value: 'x' });
});

test('Radio w/ onChange()', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: true } };

    const snap = snapshot.shallow(
        <Radio name='radio' value='x' onChange={fn}>
            Label
        </Radio>
    );

    snap.childAt(0).simulate('change', event);
    expect(snap).toMatchSnapshot();

    expect(fn).toHaveBeenCalled();
    expect(fn).toHaveBeenCalledWith(event, { name: 'radio', value: 'x' });
});

test('Radio w/ onChange() disabled', () => {
    const fn = jest.fn();
    const event = { currentTarget: { checked: true } };

    const snap = snapshot.shallow(
        <Radio name='radio' value='x' disabled onChange={fn}>
            Label
        </Radio>
    );

    snap.childAt(0).simulate('change', event);
    expect(fn).not.toHaveBeenCalled();
    expect(fn).not.toHaveBeenCalledWith(event, { name: 'radio', value: 'x' });
});
