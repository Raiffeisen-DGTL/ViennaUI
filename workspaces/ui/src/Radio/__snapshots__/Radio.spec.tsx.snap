// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio 1`] = `
<label
  class="Radiostyles__Box-oebwjf-2 OmtsL"
>
  <input
    aria-invalid="false"
    class="Radiostyles__Input-oebwjf-3 dfmTSl"
    type="radio"
    value="x"
  />
  <span
    class="Radiostyles__Icon-oebwjf-0 gysarQ"
  />
</label>
`;

exports[`Radio w/ any props 1`] = `
<label
  class="Radiostyles__Box-oebwjf-2 OmtsL className"
  id="id"
  title="title"
>
  <input
    aria-invalid="false"
    class="Radiostyles__Input-oebwjf-3 dfmTSl"
    name="name"
    type="radio"
    value="x"
  />
  <span
    class="Radiostyles__Icon-oebwjf-0 gysarQ"
  />
  <span
    class="Radiostyles__Content-oebwjf-1 bKkGza"
  >
    Label
  </span>
</label>
`;

exports[`Radio w/ checked 1`] = `
Array [
  <label
    class="Radiostyles__Box-oebwjf-2 OmtsL"
  >
    <input
      aria-invalid="false"
      checked=""
      class="Radiostyles__Input-oebwjf-3 dfmTSl"
      name="Radio1"
      type="radio"
      value="x"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 gysarQ"
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
  <label
    class="Radiostyles__Box-oebwjf-2 OmtsL"
  >
    <input
      aria-invalid="false"
      class="Radiostyles__Input-oebwjf-3 dfmTSl"
      name="Radio2"
      type="radio"
      value="y"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 gysarQ"
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Radio w/ checked and disabled 1`] = `
Array [
  <label
    class="Radiostyles__Box-oebwjf-2 npIOB"
    disabled=""
  >
    <input
      aria-invalid="false"
      checked=""
      class="Radiostyles__Input-oebwjf-3 yYQyJ"
      disabled=""
      name="Radio1"
      type="radio"
      value="x"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 rpUxC"
      disabled=""
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
  <label
    class="Radiostyles__Box-oebwjf-2 OmtsL"
  >
    <input
      aria-invalid="false"
      class="Radiostyles__Input-oebwjf-3 dfmTSl"
      name="Radio2"
      type="radio"
      value="y"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 gysarQ"
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Radio w/ disabled 1`] = `
Array [
  <label
    class="Radiostyles__Box-oebwjf-2 npIOB"
    disabled=""
  >
    <input
      aria-invalid="false"
      class="Radiostyles__Input-oebwjf-3 yYQyJ"
      disabled=""
      name="Radio1"
      type="radio"
      value="x"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 rpUxC"
      disabled=""
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
  <label
    class="Radiostyles__Box-oebwjf-2 OmtsL"
  >
    <input
      aria-invalid="false"
      class="Radiostyles__Input-oebwjf-3 dfmTSl"
      name="Radio2"
      type="radio"
      value="y"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 gysarQ"
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Radio w/ invalid 1`] = `
<label
  class="Radiostyles__Box-oebwjf-2 OmtsL"
>
  <input
    aria-invalid="true"
    class="Radiostyles__Input-oebwjf-3 bjGmrm"
    name="Radio1"
    type="radio"
    value="x"
  />
  <span
    class="Radiostyles__Icon-oebwjf-0 beTYGv"
  />
  <span
    class="Radiostyles__Content-oebwjf-1 bKkGza"
  >
    Label
  </span>
</label>
`;

exports[`Radio w/ label 1`] = `
<label
  class="Radiostyles__Box-oebwjf-2 OmtsL"
>
  <input
    aria-invalid="false"
    class="Radiostyles__Input-oebwjf-3 dfmTSl"
    type="radio"
    value="x"
  />
  <span
    class="Radiostyles__Icon-oebwjf-0 gysarQ"
  />
  <span
    class="Radiostyles__Content-oebwjf-1 bKkGza"
  >
    Label
  </span>
</label>
`;

exports[`Radio w/ name 1`] = `
Array [
  <label
    class="Radiostyles__Box-oebwjf-2 OmtsL"
  >
    <input
      aria-invalid="false"
      class="Radiostyles__Input-oebwjf-3 dfmTSl"
      name="Radio1"
      type="radio"
      value="x"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 gysarQ"
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
  <label
    class="Radiostyles__Box-oebwjf-2 OmtsL"
  >
    <input
      aria-invalid="false"
      class="Radiostyles__Input-oebwjf-3 dfmTSl"
      name="Radio2"
      type="radio"
      value="y"
    />
    <span
      class="Radiostyles__Icon-oebwjf-0 gysarQ"
    />
    <span
      class="Radiostyles__Content-oebwjf-1 bKkGza"
    >
      Label
    </span>
  </label>,
]
`;

exports[`Radio w/ onChange() 1`] = `
<Radiostyles__Box
  disabled={false}
  size="m"
>
  <Radiostyles__Input
    aria-invalid={false}
    checked={false}
    disabled={false}
    name="radio"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    type="radio"
    value="x"
  />
  <Radiostyles__Icon
    disabled={false}
    size="m"
  />
  <Radiostyles__Content
    size="m"
  >
    Label
  </Radiostyles__Content>
</Radiostyles__Box>
`;

exports[`Radio w/ size: l 1`] = `
<label
  class="Radiostyles__Box-oebwjf-2 OmtsL"
>
  <input
    aria-invalid="false"
    class="Radiostyles__Input-oebwjf-3 dfmTSl"
    type="radio"
    value="x"
  />
  <span
    class="Radiostyles__Icon-oebwjf-0 gnjoPM"
  />
  <span
    class="Radiostyles__Content-oebwjf-1 eNlTGc"
  >
    Label
  </span>
</label>
`;

exports[`Radio w/ size: m 1`] = `
<label
  class="Radiostyles__Box-oebwjf-2 OmtsL"
>
  <input
    aria-invalid="false"
    class="Radiostyles__Input-oebwjf-3 dfmTSl"
    type="radio"
    value="x"
  />
  <span
    class="Radiostyles__Icon-oebwjf-0 gysarQ"
  />
  <span
    class="Radiostyles__Content-oebwjf-1 bKkGza"
  >
    Label
  </span>
</label>
`;

exports[`Radio w/ size: s 1`] = `
<label
  class="Radiostyles__Box-oebwjf-2 OmtsL"
>
  <input
    aria-invalid="false"
    class="Radiostyles__Input-oebwjf-3 dfmTSl"
    type="radio"
    value="x"
  />
  <span
    class="Radiostyles__Icon-oebwjf-0 jiYKjX"
  />
  <span
    class="Radiostyles__Content-oebwjf-1 cpQmyB"
  >
    Label
  </span>
</label>
`;
